### PR TITLE
Add an import from ChainRules macro

### LINKED ISSUE #132: ChainRules import mechanism
Something like what [ReverseDiff has](https://github.com/JuliaDiff/ReverseDiff.jl/blob/v1.14.1/src/macros.jl#L316). The NNlib integration at https://github.com/FluxML/Tracker.jl/blob/eeb0ae8b50bd1dd02d01bcc1d0c1b6290af251bb/src/lib/array.jl#L471-L557 seems like a good target for this type of macro.

### PR DESCRIPTION
**Pull request by avik-pal**
Adds a macro to import rrules from ChainRules.

Fixes #132 

### PR Checklist

- [x] Tests are added
- [x] Documentation, if applicable

(can't seem to request reviewers directly)

cc @ToucheSir

### CODE REVIEW

File: test/chainrules.jl [hunk 1]
```diff
@@ -0,0 +1,183 @@
...
+  function back(d)
+    #=
+    The proper derivative of `f` is 4, but in order to
+    check if `ChainRulesCore.rrule` had taken over the compuation,
+    we define a rrule that returns 3 as `f`'s derivative.
+
+    After importing this rrule into Tracker, if we get 3
+    rather than 4 when we compute the derivative of `f`, it means
+    the importing mechanism works.
+    =#
```
[the remark this answers is not in the document]
Comment (avik-pal) [reply test/chainrules.jl:25]: I can change it to that as well if that is preferred.

[the remark this answers is not in the document]
Comment (oxinabox) [reply test/chainrules.jl:25]: It might be more transperent

**Review by oxinabox — APPROVED** [review]
merge once happy you have addressed these

File: test/chainrules.jl (hunk 1 above)
[the remark this answers is not in the document]
Comment (avik-pal) [reply test/chainrules.jl]: Done

File: Project.toml [hunk 2]
```diff
@@ -45,8 +47,10 @@ Statistics = "1"
...
 
 [extras]
+ChainRulesCore = "d360d2e6-b24c-11e9-a2a3-2a2ae2dbcce4"
+LinearAlgebra = "37e2e46d-f89d-539d-b4ee-838fcccc9c8e"
 PDMats = "90014a1f-27ba-587c-ab20-58faa44d9150"
 Test = "8dfed614-e22c-5e08-85e1-65c5234f0b40"
 
 [targets]
-test = ["PDMats", "Test"]
+test = ["ChainRulesCore", "LinearAlgebra", "PDMats", "Test"]
```
[the remark this answers is not in the document]
Comment (avik-pal) [reply Project.toml]: I swear I did not know this even after so many years :sweat:

**Comment by avik-pal** [comment]
@oxinabox you will have to merge, I don't have merge rights

**Pull request merged**